client/my-sites/people/role-select/index.jsx

```diff
@@ -2,138 +2,83 @@
-
-	refreshRoles( nextProps ) {
-		const siteId = nextProps && nextProps.siteId ? nextProps.siteId : this.props.siteId;
-
-		if ( siteId ) {
-			let siteRoles = RolesStore.getRoles( siteId );
-
-			if ( this.props.includeFollower ) {
-				siteRoles = Object.assign( {}, this.getWPCOMFollowerRole(), siteRoles );
-			}
-
-			this.setState( {
-				roles: siteRoles
-			} );
-		}
-	},
-
-	fetchRoles( siteId = this.props.siteId ) {
-		if ( ! siteId ) {
-			debug( 'siteId not set' );
-			return;
-		}
-
-		if ( Object.keys( RolesStore.getRoles( siteId ) ).length ) {
-			debug( 'initial fetch not necessary' );
-			return;
-		}
-
-		debug( 'Fetching roles for ' + siteId );
-
-		// defer fetch requests to avoid dispatcher conflicts
-		setTimeout( function() {
-			const fetching = RolesStore.isFetching( siteId );
-			if ( fetching ) {
-				return;
-			}
-			RolesActions.fetch( siteId );
-		}, 0 );
-	},
+import QuerySites from 'components/data/query-sites';
+import QuerySiteRoles from 'components/data/query-site-roles';
+import { getSite } from 'state/sites/selectors';
+import { getSiteRoles } from 'state/site-roles/selectors';
+
+const getWpcomFollowerRole = ( { site, translate } ) => {
+	const displayName = site.is_private
+		? translate( 'Viewer', { context: 'Role that is displayed in a select' } )
+		: translate( 'Follower', { context: 'Role that is displayed in a select' } );
+
+	return {
+		display_name: displayName,
+		name: 'follower'
+	};
+};
+
+const RoleSelect = ( props ) => {
+	let { siteRoles } = props;
+	const { site, includeFollower, siteId, id, explanation, translate } = props;
+	const omitProps = [
+		'site',
+		'key',
+		'siteId',
+		'includeFollower',
+		'explanation',
+		'siteRoles',
+		'dispatch',
+		'moment',
+		'numberFormat',
+		'translate'
+	];
+
+	if ( site && siteRoles && includeFollower ) {
+		siteRoles = siteRoles.concat( getWpcomFollowerRole( props ) );
+	}
 
-	render() {
-		const roleKeys = Object.keys( this.state.roles );
-		return (
-			<FormFieldset key={ this.props.siteId } disabled={ ! roleKeys.length }>
-				<FormLabel htmlFor={ this.props.id }>
-					{ this.translate( 'Role', {
-						context: 'Text that is displayed in a label of a form.'
-					} ) }
-				</FormLabel>
-				<FormSelect { ...omit( this.props, [ 'site', 'key', 'siteId', 'includeFollower', 'explanation' ] ) }>
-					{
-						map( this.state.roles, ( roleObject, key ) => {
-							return (
-								<option value={ key } key={ key }>
-									{ roleObject.display_name }
-								</option>
-							);
-						} )
-					}
-				</FormSelect>
-				{ this.props.explanation &&
-					<FormSettingExplanation>
-						{ this.props.explanation }
-					</FormSettingExplanation>
+	return (
+		<FormFieldset key={ siteId } disabled={ ! siteRoles }>
+			{ siteId && <QuerySites siteId={ siteId } /> }
+			{ siteId && <QuerySiteRoles siteId={ siteId } /> }
+			<FormLabel htmlFor={ id }>
+				{ translate( 'Role' ) }
+			</FormLabel>
+			<FormSelect { ...omit( props, omitProps ) }>
```

Comment:
If we could figure out based on usage what's expected for this component to pass through (e.g. `onChange`), would be nice to avoid the `omit`. I see in a few cases there's an `id` and `name` passed, though I'm doubtful those are really necessary.